src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -258,8 +258,12 @@
     /**
      * Mournign parameters.
      */
-    private boolean needToMourn = false;
-    private boolean mourning    = false;
+    private List<String> needToMourn = new ArrayList<>();
+
+    /**
+     * If the colony is currently mourning
```

Comment:
I'd do this differently, on mourning, if the grave is at the graveyard, go to the graveyard to the grave, on arrival, double the chance to stop mourning

Reply:
I didn't wanted to refactor the complete mourning system. Before the mourning system was just a bool set to true for the whole colony. I saw somewhere that there is a plan-ish feature to change it and add relationship etc. (not sure)

I can change it to had a morning timer per dead citizen and decrease this timer when the grave is restore and add a chance system. But for now, there is no chance system at all - even before this PR

Reply:
You have to change even less, the mourning AI already has a counter with a probability of stopping to mourn, you should be able to adjust it there easily.

Reply:
@Raycoms For this PR I would consider this fine. We can overhaul it later then.

Reply:
Hem, I didn't think about the Mourning AI indeed. That will be great to change it later so the mourning citizen visit the grave etc. I can work on that in the future. Let me know if it's absolutely required for this PR.

Reply:
Okay, let's do it like this then, to make this PR not too big.
Remove the current mourning changes.
And then in a follow-up PR we adjust the mourning AI based on the graves.

Reply:
will do